### DISCUSSION TITLE
Weekly Stand-Up Discussion - Sprint #2

### DISCUSSION THREAD

**Discussion by Rankin-Craig** [JamesSG2/Handwriting-Conversion-Application#10] [General]
A Stand-Up Discussion or meeting is something that would take place on a daily basis in a real-world software design team setting.  It is called a "Stand-Up meeting" because it is meant to be done very quickly to inform your other group members about the work you have been doing.  Since we are attempting to mimic this situation we will complete a Stand-Up discussion every week. 

Please answer the following questions regarding the work you have done from Friday 3/19 until Wednesday 3/31:
  

1. What specific tasks did you work on during this time period (Sprint #2)?  Explain each one.
2.   What task did you spend the most time on?
3.   Did you run into any problems or roadblocks that you may need to discuss with the group or others? - Describe those issues.
4.   Evaluate the responses in this discussion board from others.  Respond by offering your help or advice to someone else.

**Comment by rover-t** [comment]
1. I worked on: creating a system to input image files in a consistent format, determining the best methods for creating a handwriting sample, and building a spellchecker for error correction. I've also been working on making my syntax more "pythonic" with a clearer structure.

2. I spent the most time on the spell checker. I was frustrated by close but not quite errors e.g. "qulck" instead of "quick." I created a function that takes a string representing what the image should says, and corrects errors in character recognition.

3. I've been trying to get an effective way to identify punctuation characters. Tesseract does a very poor job of this, so I'm trying make a workaround where you place punctuation characters in a series of boxes in a predetermined order. Then I can implement a recognition system that identifies the position of the colored boxes and captures the character contained inside.

**Comment by JamesSG2** [comment]
1.  I worked on:
- Organizing the GitHub repository a bit; Created a Main Development branch (dev) in which newly created features should be merged to, then tested and worked out prior to merging with the "master" branch which should always be a stable 'final' product (I renamed the main and other branches to follow naming conventions) 
- Ran both Trevor and Antoni's code and was able to get everything set up and successfully compile with the expected output

2.
- Researching GitHub repository conventions and trying to get an idea for the kind of structure we want to have
- Researching how to get words broken into individual letters more effectively

3. / 4.
I spent much of my time dealing with errors (files in the wrong spot or fixing disagreements with different versions, etc.) while trying to get the code from the master branch to run. I was only able to get Antoni's code to run by telling the program exactly where to find pytesseract on my PC -- not sure how to make that more generalized. I think more detailed instructions for all the exact programs and versions that were used would be helpful in case other (or new) people want to run the code (or simply uploading those programs directly to this repository if possible).

**Reply by rover-t** [reply]
I noticed you renamed the main branch to master. Is it possible to change it back to main, or something else like primary. It's not considered good practice anymore to name things using aspects of the master/slave terminology. You can avoid the hassle of giving the location of pytesseract every time by adding pytesseract to PATH. I can give some explanation on that during the meet.

**Reply by JamesSG2** [reply]
K, could you send a link or something that shows/explains the branching terminology you're talking about? The article I used was quite recent, especially the official GitHub tutorial which also uses 'master'. I'll switch it back though in case there are any compatibility issues with the current code. Thanks, and I'll work on getting pytesseract set up on PATH so I can work with the code more easily.

**Reply by Rankin-Craig** [reply]
Here is an article discussing the issue...
https://technology.inquirer.net/101012/github-to-change-master-and-slave-coding-terms-deemed-as-oppressive-metaphors-racially-inappropriate#:~:text=Microsoft%2Downed%20computer%20code%20company,American%20George%20Floyd%20last%20May.

**Reply by JamesSG2** [reply]
Oh, ok. Thanks, I had no idea. I thought it was just a technical issue or incompatibility

**Reply by Rankin-Craig** [reply]
Until Trevor mentioned it - I hadn't thought about that issue either.  Its also an issue in the IT world - many computers use dual hard drives that are designated as master/slave.  Good catch Trevor!!

**Comment by BigWub67** [comment]
1. I started to look into ways of splitting a image of a sentence into images of a word.

2. I only started the one thing.

3. Not yet
